tracing-attributes/src/expand.rs

```diff
@@ -440,8 +440,12 @@ fn param_names(pat: Pat, record_type: RecordType) -> Box<dyn Iterator<Item = (Id
 enum AsyncTraitKind<'a> {
     // old construction. Contains the function
     Function(&'a ItemFn),
-    // new construction. Contains a reference to the async block
-    Async(&'a ExprAsync),
+    // new construction. Contains a reference to the async block, and a bool
+    // indicating if the return value is a `Box::pin`
+    Async {
+        async_expr: &'a ExprAsync,
+        pinned_box: bool,
+    },
 }
```

Comment:
i'm not sure if this should still be called `AsyncTraitKind`. since `#[instrument]` now handles _all_ functions returning async blocks, not just those generated by the `#[async_trait]` macro, i think it would be nice to rename this type (and other types) to reflect that.